### PR TITLE
[Feature] Renewal Form API Hookup

### PR DESCRIPTION
**Pull request by OustanDing**
## Notion ticket link

[Ticket](https://www.notion.so/uwblueprintexecs/Renewal-Form-API-hookup-53df8f8b1fd94abe8139f8c693f967e0)

## Implementation description
* Renewal Form API hookup
* New `createRenewalApplication` GQL mutation for creating renewals (called when user submits form)
* New `verifyIdentity` GQL mutation for verifying user info prior to submitting renewal request - also updates the `acceptedTOS` timestamp for the applicant
* Added field resolver `applicantActivePermitResolver` for getting the currently active permit of an applicant
* Small DB changes
  * Renamed `acceptedTOC` to `acceptedTos` (terms of service) in `applicant` model
  * Separated `physicianName` to `physicianFirstName` and `physicianLastName` in `application` model
* Added `Request` state container (Unstated Next) for storing application-related state data, including the timestamp of the TOS acceptance and the applicant ID after the identity verification step

## Notes
* Have a DB relation issue that this is blocked on: We need to enforce a 1-on-1 relationship between applicants and medical information records (also applicants and guardian records) - **BLOCKED**
* Shopify payment flow isn't set up yet - we will submit the application upon clicking the "Submit" button in the form instead - need to integrate Shopify API webhooks when that is set up (future ticket)
* Added error modals where needed for invalid API errors - will address with designers in future

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

File: pages/verify-identity.tsx [hunk 1]
```diff
@@ -67,16 +146,24 @@ export default function IdentityVerificationForm() {
...
               </Link>
-              <Link href="/renew">
-                <Button
-                  variant="solid"
-                  disabled={!userId || !phoneNumberSuffix || !dateOfBirth}
-                >{`Continue`}</Button>
-              </Link>
+              <Button
+                variant="solid"
+                disabled={loading || !userId || !phoneNumberSuffix || !dateOfBirth}
```
Comment (jeffreyzhang2001) [inline pages/verify-identity.tsx:163]: Do we need loading on this as well?

File: lib/applications/resolvers.ts [hunk 2]
```diff
@@ -64,3 +68,161 @@ export const createApplication: Resolver = async (_, args, { prisma }) => {
...
+        // TODO: Modify logic when DB schema gets changed (medicalInfo is not undefined)
+        disability: applicant.medicalInformation?.disability || 'Placeholder disability',
+        physicianFirstName:
+          updatedPhysician && physicianFirstName
+            ? physicianFirstName
+            : physician?.firstName || 'Placeholder physician first name',
+        physicianLastName:
+          updatedPhysician && physicianLastName
+            ? physicianLastName
+            : physician?.lastName || 'Placeholder physician last name',
```
Comment (jeffreyzhang2001) [inline lib/applications/resolvers.ts:174]: NOTE: This is going to change with the change that combines physician's firstName and lastName into one name field.

**Review by jeffreyzhang2001 — CHANGES_REQUESTED** [review]

**Review by jeffreyzhang2001 — APPROVED** [review]

File: prisma/schema.sql [hunk 3]
```diff
@@ -171,7 +171,8 @@ CREATE TABLE applications (
   cannot_walk_100m         BOOLEAN NOT NULL DEFAULT false,
   aid                      Aid ARRAY,
   -- Physician information
-  physician_name            VARCHAR(255) NOT NULL,
+  physician_first_name      VARCHAR(255) NOT NULL,
+  physician_last_name       VARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (OustanDing) [reply prisma/schema.sql]: Resolved

File: pages/verify-identity.tsx (hunk 1 above)
Comment (OustanDing) [reply pages/verify-identity.tsx:163]: Yup, we need to call the API to verify the user's identity

**Pull request merged**